dex/src/main/scala/com/wavesplatform/dex/api/http/routes/MatcherApiRoute.scala

```diff
@@ -1033,6 +1037,55 @@ class MatcherApiRoute(
     complete { Json.toJson(orderDb.transactionsByOrder(orderId)) }
   }
 
+  @Path("/debug/config")
+  @ApiOperation(
+    value = "Returns current matcher's configuration",
+    httpMethod = "GET",
+    authorizations = Array(new Authorization(SwaggerDocService.apiKeyDefinitionName)),
+    tags = Array("debug"),
+    response = classOf[HttpResponse]
+  )
+  def getConfig: Route = (path("config") & get & withAuth)  {
+    complete {
+
+      var outputConfig = config.getObject("waves").toConfig
+
+      def pathShouldNotBeDeleted(path: String): Boolean = ("(user|pass|seed|private)".r findFirstIn path).isEmpty
+
+      def modifyConfig(path: String, config: Config): Unit = {
+        val itr = config.root().entrySet().iterator()
+
+        while(itr.hasNext()){
```

Comment:
The `while` expression is used in Scala only for a highly optimized code. In this task it could be used for Breadth First Search. But we haven't thousands of keys, so we can use a brute-force solution like this code.

Also iterators aren't used often in Scala, because this code is error prone (e.g. running `next` twice or something else). So it'd better to use `import scala.jdk.CollectionConverters.MapHasAsScala` and `asScala.foreach` instead of `iterator`

Reply:
changed to `asScala.foreach`